src/clarity-angular/input-trap/input-trap.directive.ts

```diff
@@ -0,0 +1,47 @@
+import { Directive, HostListener, ElementRef, OnInit } from "@angular/core";
+
+@Directive({
+    selector: "[clrInputTrap]"
+})
+export class InputTrapDirective implements OnInit {
+
+    static TAB_KEY = 9;
+
+    // tslint:disable-next-line:max-line-length
+    // reference: https://github.com/jkup/focusable
+    static FOCUSABLE_INPUTS = "a[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled])";
+
+    firstInput: HTMLElement;
+    lastInput: HTMLElement;
+
+    constructor(public elementRef: ElementRef) { }
+
+    @HostListener("document:keydown", ["$event"])
+    onkeydown(event: any) {
```

Comment:
@youdz Will go ahead and change to use the `focusin` event. However, don't we need to still determine what is considered the first and last focusable elements? Would I expand the list to include those with `tabindex`? Then that would include more logic, right? Since we then have to take into account those indices.

Just want to make sure I understand everything here 😄